CONTRIBUTING.md

```diff
@@ -0,0 +1,113 @@
+## How to Contribute
+
+The OCS-Operator project is under the [Apache 2.0 license](LICENSE). We accept
+contributions via GitHub pull requests. This document outlines how to
+contribute to the project.
+
+## Contribution Flow
+
+Developers must follow these steps to make a change:
+
+1. Fork the `openshift/ocs-operator` repository on GitHub.
+2. Create a branch from the master branch, or from a versioned branch (such
+   as `release-4.2`) if you are proposing a backport.
+3. Make changes.
+4. Create tests as needed and ensure that all tests pass.
+5. Ensure your commit messages include sign-off.
+6. Push your changes to a branch in your fork of the repository.
+7. Submit a pull request to the `openshift/ocs-operator` repository.
+8. Work with the community to make any necessary changes through the code
+   review process (effectively repeating steps 3-8 as needed).
+
+## Developer Environment Installation
+
+Instructions to create a dev environment for OCS-operator can be found in the
+[main project documentation](./README.md#Installation of development builds).
+
+## Commit Requirements
+
+In addition to passing [functional tests](./README.md#functional-tests), all
+commits to OCS-operator must follow the guidelines in this section. You can
+verify your changes meet most of these standards by running the following
+command in the repository's root directory:
+
+```
+make ocs-operator-ci
+```
+
+### Commits Per Pull Request
+
+*This requirement cannot be tested by make ocs-operator-ci.*
+
+OCS-operator is a project which maintains several versioned branches
+independently. When backports are necessary, monolithic commits make it
+difficult for maintainers to cleanly backport only the necessary changes.
+
+Where possible, pull requests should be composed of multiple commits that
+each make small but meaningful changes. Striking a balance between minimal
+commits and logically complete changes is an art as much as a science, but
+when it is possible and reasonable, divide your pull request into more commits.
+
+### Commit and Pull Request Messages
+
+*This requirement cannot be fully tested by make ocs-operator-ci.*
+
+All commit messages must contain:
+- A single summary line describing the change
+- A single blank line
+- A longer description of the change, including:
+  - Why the change is necessary or useful
+  - How the change was implemented
+  - A reference to any open Issues the change addresses
+- A sign-off, as discussed in [Certificate of Origin](#certificate-of-origin).
```

Comment:
blank line before the signoff line

Reply:
I suggest we also include an optional `Co-authored-by:` line.

Reply:
Hm; interesting. I'm now wondering if in the Co-authored-by: case a signoff is also necessary. Given my reading of the DCO, I would have to imagine signoffs by all authors would be safer legally, so I'm inclined to require that as well. @obnoxxx, @raghavendra-talur, agreed on that?

Reply:
There's a little bit of discussion on that here: https://stackoverflow.com/questions/7442112/how-to-attribute-a-single-commit-to-multiple-developers

Co-authored-by seems to be a pure GitHub construction, but there's no technical specification in the DCO for having a "Signed-off-by" string in the commit message. I think it was more meaningful when there was a strict requirement of GPG signing commits. I would say we can treat Co-authored-by as an implicit sign-off, since it fulfills the requirement of recording the contribution and has the implication that the co-author also meets the legal requirements.

Reply:
> Hm; interesting. I'm now wondering if in the Co-authored-by: case a signoff is also necessary. Given my reading of the DCO, I would have to imagine signoffs by all authors would be safer legally, so I'm inclined to require that as well. @obnoxxx, @raghavendra-talur, agreed on that?

Yes, usually I would expect sign-off lines by all authors.
In the samba project, we use "Pair-Programmed-With: " instead of "Co-Authored-By: ". But the real requirement is a sign-off line for each author.